include/translatable-object.php

```diff
@@ -246,53 +246,110 @@ public function delete_language( $id ) {
 	}
 
 	/**
-	 * Wraps `wp_get_object_terms()` to cache it and return only one object.
-	 * Inspired by the WordPress function `get_the_terms()`.
+	 * Wraps wp_get_object_terms() to cache it for multiple objects.
+	 * Similar to get_object_term() but for multiple objects.
 	 *
-	 * @since 1.2
+	 * @since 3.8
 	 *
-	 * @param int    $id       Object ID.
-	 * @param string $taxonomy Polylang taxonomy depending if we are looking for a post (or term, or else) language.
-	 * @return WP_Term|false The term associated to the object in the requested taxonomy if it exists, `false` otherwise.
+	 * @param int[]  $object_ids Array of object IDs.
+	 * @param string $taxonomy   Taxonomy name.
+	 * @return array<int,WP_Term|null> Array of terms with object ID as key.
 	 */
-	public function get_object_term( $id, $taxonomy ) {
-		$id = $this->sanitize_int_id( $id );
+	protected function get_object_terms( array $object_ids, string $taxonomy ) {
+		$object_ids = $this->sanitize_int_ids_list( $object_ids );
+		if ( empty( $object_ids ) ) {
+			return array();
+		}
 
-		if ( empty( $id ) ) {
-			return false;
+		$this->update_object_term_cache( $object_ids );
+
+		$cached_values = wp_cache_get_multiple( $object_ids, "{$taxonomy}_relationships" );
+
+		// Flatten the array to prime the terms cache.
+		$all_term_ids = array();
+		foreach ( $cached_values as $term_ids ) {
+			$all_term_ids = array_merge( $all_term_ids, $term_ids );
+		}
+		_prime_term_caches( $all_term_ids, false );
+
+		$terms = array();
+		foreach ( $cached_values as $object_id => $term_ids ) {
+			if ( ! empty( $term_ids ) ) {
+				$term_id             = reset( $term_ids ); // There is only one term for language or translation groups.
+				$term                = get_term( $term_id );
+				$terms[ $object_id ] = $term instanceof WP_Term ? $term : null;
```

Comment:
For which usage do we need to store null values? (apart the fact that I don't see how it could occur as I expect `$term_id` to always be associated to a valid term).